### PR TITLE
Fix aggregation failing if index does not exists

### PR DESCRIPTION
**Pull request by dvvolynkin**

### CODE REVIEW

**Review by pcorpet — CHANGES_REQUESTED** [review]
Thanks for the fix.

Please add a test in test__mongomock to ensure that there are no regression and that we have the same behavior than pymongo going forward.

**Comment by dvvolynkin** [comment]
Done

File: mongomock/aggregate.py [hunk 1]
```diff
@@ -315,7 +315,16 @@ def _handle_project_operator(self, operator, values):
         if operator == '$arrayElemAt':
             key, index = values
             array = self._parse_basic_expression(key)
-            return array[index]
+
+            if index < len(array):
+                return array[index]
```
Comment (pcorpet) [inline mongomock/aggregate.py:376]: What about negative values? Can you add a test for that as well please?

I guess the implementation could be easier and more pythonic with:
```py
try:
  return array[index]
except IndexError:
  raise KeyError('%s is not a valid index in array' % index)
```

File: mongomock/aggregate.py [hunk 2]
```diff
@@ -315,7 +315,16 @@ def _handle_project_operator(self, operator, values):
...
+            # As I understood, this type of exception will be caught in the top
```
Comment (pcorpet) [inline mongomock/aggregate.py]: It's not slower: that's great, you can get rid of the comment

**Review by pcorpet — CHANGES_REQUESTED** [review]

**Review by pcorpet — APPROVED** [review]

**Comment by pcorpet** [comment]
Thanks Daniil

**Pull request merged**